src/tracing/agent.h

```diff
@@ -31,24 +32,51 @@ class TracingController : public v8::platform::tracing::TracingController {
   }
 };
 
+class Agent;
 
-class Agent {
+// Allows agent to be destroyed early. Methods are noop after the agent was
+// destroyed
+class AgentHandle {
  public:
-  // Resetting the pointer disconnects client
-  using ClientHandle = std::unique_ptr<std::pair<Agent*, int>,
-                                       void (*)(std::pair<Agent*, int>*)>;
+  explicit AgentHandle(Agent*);
```

Comment:
`Agent* agent` and consider changing the data member to `Agent* const agent_;`

Reply:
See comment above.